components/console-backend-service/internal/domain/k8s/namespace_resolver.go

```diff
@@ -95,3 +103,29 @@ func (r *namespaceResolver) ApplicationsField(ctx context.Context, obj *gqlschem
 
 	return appNames, nil
 }
+
+func (r *namespaceResolver) CreateNamespace(ctx context.Context, name string, labels *gqlschema.Labels) (gqlschema.NamespaceCreationOutput, error) {
+
+	// TODO: remove the 'env' label when the method to distinguish if it's a system namespace or not is changed
+	gqlLabels := r.setLabels(labels)
+	_, err := r.namespaceSvc.Create(name, gqlLabels)
+	if err != nil {
+		glog.Error(errors.Wrapf(err, "while creating %s `%s`", pretty.Namespace, name))
+		return gqlschema.NamespaceCreationOutput{}, gqlerror.New(err, pretty.Namespace, gqlerror.WithName(name))
+	}
+	return gqlschema.NamespaceCreationOutput{
+		Name:   name,
+		Labels: gqlLabels,
+	}, nil
+}
+
+func (r *namespaceResolver) setLabels(givenLabels *gqlschema.Labels) map[string]string {
```

Comment:
Rather `populate`. `Set` means that you want to set labels on `namespaceResolver`